src/Microsoft.Health.Fhir.Core/Extensions/RawResourceElementExtensions.cs

```diff
@@ -22,21 +31,21 @@ public static class RawResourceElementExtensions
         /// </summary>
         /// <param name="rawResource">Input RawResourceElement to convert to a JsonDocument</param>
         /// <param name="outputStream">Stream to serialize to</param>
-        public static async Task SerializeToStreamAsUtf8Json(this RawResourceElement rawResource, Stream outputStream)
+        /// <param name="pretty">Identifies whether to indent or not the json response</param>
+        public static async Task SerializeToStreamAsUtf8Json(this RawResourceElement rawResource, Stream outputStream, bool pretty = false)
         {
             EnsureArg.IsNotNull(rawResource, nameof(rawResource));
 
+            var jsonDocument = JsonDocument.Parse(rawResource.RawResource.Data);
```

Comment:
While this code is simpler, the motive for the previous code is that the resource wouldn't not need to be parsed on the way out if the Metadata is already set